web/web.go

```diff
@@ -111,6 +112,8 @@ func NewWebService(o *WebServiceOptions) *WebService {
 		router.Post("/-/quit", ws.quitHandler)
 	}
 
+	router.Get("/federate", instr("federate", o.Federation))
```

Comment:
Say we bump the API to v2 - this will be a collective fix insufficiencies of API v1 endpoints we observed over time.
It's highly unlikely that these decisions have any relation to the federation format.

So we drag `/api/v1/federate` to `/api/v2/federate` without changes.
In the next iteration we change the federation format, bump to API v3 and drag all other endpoints to `/api/v3/*` as they are.

So yes to versioning but no to it being part of the general API, in my opinion.
Changes to federation are also more likely to be breaking while API endpoints can be served in parallel.